src/kvstore/Part.cpp

```diff
@@ -243,6 +253,19 @@ bool Part::preProcessLog(LogID logId,
             << ", termId " << termId
             << ", clusterId " << clusterId
             << ", log " << log;
+    if (!log.empty()) {
+        switch (log[sizeof(int64_t)]) {
+            case OP_ADD_LEARNER: {
+                auto learner = decodeLearner(log);
```

Comment:
here log is const string&, but "HostAddr decodeLearner(folly::StringPiece encoded)" parameter is folly::StringPiece, why do the convert?

Reply:
Good catch.